Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -501,13 +501,15 @@ public async Task RunStartupTasksAsync()
 
             var stopWatch = new Stopwatch();
             stopWatch.Start();
-            await Task.WhenAll(StartEntryPoints(entryPoints, true)).ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested();
+            await Task.WhenAny(Task.WhenAll(StartEntryPoints(entryPoints, true)), Task.Delay(-1, cancellationToken)).ConfigureAwait(false);
```

Comment:
Why wrap `Task.WhenAll` in a `Task.WhenAny`?

Reply:
Now we can cancel it with the `cancellationToken`
(My first idea to fix the race condition)
Now that I'm looking at it, not a fan.